src/Misc/tools.ts

```diff
@@ -1046,11 +1048,33 @@ export class Tools {
     public static readonly PerformanceConsoleLogLevel = 2;
 
     private static _performance: Performance;
+    private static _nativeCounters: Map<string, unknown> | undefined;
+    private static _nativeStartPerformanceCounter: ((name: string) => unknown) | undefined;
+    private static _nativeEndPerformanceCounter: ((counter: unknown) => void) | undefined;
 
     /**
      * Sets the current performance log level
      */
     public static set PerformanceLogLevel(level: number) {
+        {
+            let nativeEnablePerformanceLogging: (() => void) | undefined;
+            let nativeDisablePerformanceLogging: (() => void) | undefined;
+            if (typeof _native !== 'undefined') {
+                nativeEnablePerformanceLogging = _native.enablePerformanceLogging;
+                nativeDisablePerformanceLogging = _native.disablePerformanceLogging;
+
+                if (level === Tools.PerformanceNoneLogLevel) {
+                    if (nativeDisablePerformanceLogging) {
+                        nativeDisablePerformanceLogging();
+                    }
+                } else {
+                    if (nativeEnablePerformanceLogging) {
+                        nativeEnablePerformanceLogging();
+                    }
+                }
+            }
+        }

```

Comment:
I was thinking you would override this function in the constructor of nativeEngine.ts. Does that work?

Reply:
Would be great I was about to propose the same :-)